### ISSUE TITLE
Set focus styles on flat Buttons

### ISSUE THREAD

**Issue by woutsluiter** [MyOnlineStore/bricks#313]
### Bricks version 📦
`0.9.2`

### Describe the bug 🔍
When global css properties are inherited on the focus state of a `<Button variant="plain" />`. Seems that `FlatButtonStylesType` doesn't have a focus state.

### To Reproduce 💥
- Create global focus styling on a `button` element
- Add a `<Button variant="plain" />`
- Focus

### Expected behavior ✨
The styles on a focus state are set as well.

### Screenshots 📸
![image](https://user-images.githubusercontent.com/10552400/51107090-1b594000-17ee-11e9-92aa-3dab3c5a235e.png)

### Context👨‍💻
 - Device: desktop
 - OS: OSX
 - Browser Chrome
 - Browser version 71.0.3578.98

**Comment by CarloPalinckx** [comment]
Since this can't be fixed without a breaking fix, and the problem should no longer occur in the upcoming version I'm closing this.

**Comment by woutsluiter** [comment]
It actually still a thing in the 1.x branch. The IconButton has no focus styling and inherits common globally styled properties like `background` on hover and active states. I would be happy to make a PR for this. Could be something like the way it's done in `Button`:

```
type CommonType = {
    backgroundColor: string;
    color: string;
};

type ComponentStateTypes = {
    idle: CommonType;
    hover: CommonType;
    focus: CommonType;
    active: CommonType;
};

type IconButtonThemeType = {
    primary: ComponentStateTypes
    destructive: ComponentStateTypes
};
```

**Comment by woutsluiter** [comment]
Resolved in #330
